user-interface/src/main/java/life/qbic/datamanager/views/general/spreadsheet/Spreadsheet.java

```diff
@@ -211,6 +212,22 @@ public List<T> getData() {
         .toList();
   }
 
+
+  /**
+   * Get the data in the last row of the spreadsheet
+   * @return the underlying data of the last row
+   */
+  public T getLastRowData() {
+    int lastRowIndex = rowCount() - 1;
+    Row lastRow = getRow(lastRowIndex);
+    if(lastRow instanceof DataRow dataRow) {
+      return dataRow.data;
+    }
+    return null;//other possibilities?
+  }
```

Comment:
I think this method is to specific. You can get the last element of getData() instead. Alternatively you could provide a method to get the data for a specific row. In this case however there is the problematic of what is a row? do you count the header row(s)? Does it start with 0 or 1 ?
Using getData is the simplest IMHO.

Reply:
I was not sure if getData() provides ordered data - if it does I will change it accordingly.